File: src/programs/conditionalsAndLoops/one.js

import '../../styles/programs.css'
import '../../styles/topics.css'
import ConditionalsAndLoops from '../../topics/ConditionalsAndLoops'
import { useState,useEffect } from 'react'
import ConditionalsAndLoopsTwo from './two'
const ConditionalsAndLoopsOne = ({qn,backPage}) =>{
   const q2 = "Subtract the Product and Sum of Digits of an Integer"
    const back = (page) =>{
        backPage(page)
    }
        const url = window.location.href

        const [file,setFile] = useState("")
    
        useEffect(()=>{
        async function fetchFile() {
            try{
        const res = await fetch(`${url}javaPrograms/conditionalsAndLoops/One.java`)
   
        const cont = await res.text();
        
        setFile(cont)
        }
        catch(err){
        console.log(err);
        }

    }
     
    fetchFile() 
  },[])
    return(<div className="program-body">
        <div className='topics-logo'>{qn}</div>
        <div className='program-content'>

          
            <div className='program-hint program'>
                <pre>  {file}</pre>
         

            </div>

            <div onClick={() => back(<ConditionalsAndLoops/>)} className='program-btn'>Go To Map</div>
            <div onClick={() => back(<ConditionalsAndLoopsTwo qn ={q2} backPage={backPage}/>)} className='program-btn'>Next</div>
        </div>
    
    </div>)
}
export default ConditionalsAndLoopsOne

File: public/javaPrograms/conditionalsAndLoops/One.java

package javaPrograms.conditionalsAndLoops;
import java.util.*;
public class One {
    public static void main(String[] args) {
      Scanner s = new Scanner(System.in);
      System.out.print("Enter limit :");
      int limit = s.nextInt();
      int a =0;
      int b = 1;
      System.out.print(b+",");
      for (int i = 0;i<limit;i++){
        int c = a+b;
        a = b;
        b =c;
        System.out.print(c+",");
      }

    }
}
